src/battle_ai_switch_items.c

```diff
@@ -228,44 +217,177 @@ static bool8 FindMonThatAbsorbsOpponentsMove(void)
+                )
+                switchMon = FALSE;
+
+            //Check if Active Pokemon evasion boosted and might be able to dodge until awake
+            if (gBattleMons[gActiveBattler].statStages[STAT_EVASION] > (DEFAULT_STAT_STAGE + 3)
+                && AI_GetAbility(opposingBattler) != ABILITY_UNAWARE
+                && AI_GetAbility(opposingBattler) != ABILITY_KEEN_EYE
+                && !(gBattleMons[gActiveBattler].status2 & STATUS2_FORESIGHT)
+                && !(gStatuses3[gActiveBattler] & STATUS3_MIRACLE_EYED))     
+                switchMon = FALSE;
+
+        }
+
+        //Secondary Damage
+        if (monAbility != ABILITY_MAGIC_GUARD
+            && !AiExpectsToFaintPlayer())
+        {
+            //Toxic
+            moduloChance = 2; //50%
+            if (gBattleMons[gActiveBattler].status1 & (STATUS1_TOXIC_COUNTER > 2)
+                && gBattleMons[gActiveBattler].hp >= (gBattleMons[gActiveBattler].maxHP / 3)
+                && (Random() % (moduloChance*chanceReducer)) == 0)
+                switchMon = TRUE;
+            
+            //Cursed
+            moduloChance = 2; //50%
+            if (gBattleMons[gActiveBattler].status2 & STATUS2_CURSED
+                && (Random() % (moduloChance*chanceReducer)) == 0)
+                switchMon = TRUE;
+
+            //Nightmare
+            moduloChance = 3; //33.3%
+            if (gBattleMons[gActiveBattler].status1 & (STATUS1_SLEEP > 1) && gBattleMons[gActiveBattler].status2 & STATUS2_NIGHTMARE
+                && (Random() % (moduloChance*chanceReducer)) == 0)
+                switchMon = TRUE;
+
+            //Leech Seed
+            moduloChance = 4; //25%
+            if (gStatuses3[gActiveBattler] & STATUS3_LEECHSEED
+                && (Random() % (moduloChance*chanceReducer)) == 0)
+                switchMon = TRUE;
+        }
+
+        //Infatuation
+        if (gBattleMons[gActiveBattler].status2 & STATUS2_INFATUATION
+            && !AiExpectsToFaintPlayer())
+            switchMon = TRUE;
+
+        //Todo
+        //Pass Wish Heal
+        
+        //Semi-Invulnerable
+        if (gStatuses3[opposingBattler] & STATUS3_SEMI_INVULNERABLE)
+            if (FindMonThatAbsorbsOpponentsMove()) //If find absorber default to switch
+                switchMon = TRUE;
+            if (!AI_OpponentCanFaintAiWithMod(0)
+                && AnyStatIsRaised(gActiveBattler))
+                switchMon = FALSE;
+            if (AiExpectsToFaintPlayer()
+                && GetAIChosenMove(gActiveBattler) == AI_IS_SLOWER
+                && !AI_OpponentCanFaintAiWithMod(0))
+                switchMon = FALSE;
     }
-    else if (gBattleMoves[gLastLandedMoves[gActiveBattler]].power == 0
-          && Random() & 1)
+
+
+    if (switchMon)
     {
         *(gBattleStruct->AI_monToSwitchIntoId + gActiveBattler) = PARTY_SIZE;
         BtlController_EmitTwoReturnValues(BUFFER_B, B_ACTION_SWITCH, 0);
         return TRUE;
     }
+    else
```

Comment:
pret style is that if braces are around any part of an if/else statement. braces go around every if and/or else statement, so:
```
else
{
    return FALSE;
}
```